spec/lib/extensions/ar_virtual_spec.rb

```diff
@@ -434,6 +434,30 @@ class TestSubclass < TestClass
       end
     end
 
+    describe "#virtual_has_many" do
+      it "use collect to generate ids" do
+        c = Class.new(TestClassBase) do
+          virtual_has_many(:hosts)
+          def hosts
+            [OpenStruct.new(:id => 5), OpenStruct.new(:id => 6)]
+          end
+        end.new
+
+        expect(c.host_ids).to eq([5, 6])
+      end
+
+      it "uses ids to generate ids" do
```

Comment:
Can we word these at a slightly higher level? This one is about supporting Relation#ids, and the other is about, for an array, falling back to calling `id`. I'm struggling to come up with a suitably succinct alternative string, but maybe I've inspired you?